src/test/java/org/jitsi/meet/test/DialInAudioTest.java

```diff
@@ -0,0 +1,172 @@
+ * kick it and consider this test successful.
+ *
+ * @author Damian Minkov
+ */
+public class DialInAudioTest
+    extends WebTestBase
+{
+    /**
+     * Name of the system property which holds the url to execute which will
+     * create a dial-in participant.
+     * The url must be a valid url, with properly escaped parameters.
+     * The url can contain {0}, to pass the conference pin.
+     * For example: {"pin":"{0}"}, needs to be: %7B%22pin%22%3A%22{0}%22%7D
+     */
+    private static final String DIAL_IN_PARTICIPANT_REST_URL
+        = "dialIn.rest.url";
+
+    private String dialInPin = null;
+
+    private String restURLString = null;
+
+    @Override
+    public boolean skipTestByDefault()
+    {
+        return true;
+    }
+
+    /**
+     * Creates a web participant to enter a room and read the dial-in info.
+     */
+    @Test
+    public void enterAndReadDialInPin()
+    {
+        this.restURLString = System.getProperty(DIAL_IN_PARTICIPANT_REST_URL);
+        Assert.assertTrue(
+            this.restURLString != null,
+            "REST Url missing. Pass it using " +
+                "-D" + DIAL_IN_PARTICIPANT_REST_URL + "=");
+
+        // join in a room
+        ensureOneParticipant();
+
+        WebParticipant participant = getParticipant1();
+
+        if (!MeetUtils.isDialInEnabled(participant.getDriver()))
+        {
+            throw new SkipException(
+                "No dial in configuration detected. Disabling test.");
+        }
+        InfoDialog infoDialog = participant.getInfoDialog();
+        infoDialog.open();
+
+        // get the dial-in pin
+        String displayedPin = infoDialog.getPinNumber();
+        assertTrue(displayedPin.length() > 1);
+
+        infoDialog.close();
+
+        // we need to remove the last character from the pin which is '#'
+        dialInPin = displayedPin.substring(0, displayedPin.length() - 1);
+    }
+
+    /**
+     * Executes the rest configured web call, replacing the dial-in pin the url
+     * and waits 10 seconds for the participant to join.
+     */
+    @Test(dependsOnMethods = { "enterAndReadDialInPin" })
+    public void enterDialInParticipant()
+    {
+        try
+        {
+            URI restURI = new URI(
+                MessageFormat.format(
+                    this.restURLString,
+                    this.dialInPin));
+
+            RequestConfig defaultRequestConfig = RequestConfig.custom()
+                .setSocketTimeout(5000)
+                .setConnectTimeout(5000)
+                .setConnectionRequestTimeout(5000)
+                .build();
+            CloseableHttpClient httpclient = HttpClientBuilder.create()
+                .setDefaultRequestConfig(defaultRequestConfig).build();
+
+            HttpHost targetHost = new HttpHost(
+                restURI.getHost(),
+                restURI.getPort(),
+                restURI.getScheme());
+
+            HttpGet httpget = new HttpGet(restURI);
```

Comment:
For the sake of page objects, should this request logic be hidden behind a method on the participant or even in the infodialog? I was thinking the ideal is for the tests to read like english as much as possible and for test logic to be hidden away somewhere else.

Reply:
Well, this is very specific for the current test. Initially had even named the test with REST in the name...